### PR TITLE
[SYCL][XPTI] Report memory allocation info from SYCL runtime

### PR DESCRIPTION
**Pull request by alexbatashev**
This patch adds new types of notifications for XPTI to support custom memory allocators. Scoped events allow tools to ignore whatever logic is between begin and end notifications, and just assume, that the memory has been allocated. This is useful when application uses pooling to allocate memory, since while there's free space in the pool, no malloc-like calls are performed.

### CODE REVIEW

**Comment by alexbatashev** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/637

File: sycl/source/detail/memory_manager.cpp [hunk 1]
```diff
@@ -9,17 +9,106 @@
...
+  (void)AllocPtr;
+#ifdef XPTI_ENABLE_INSTRUMENTATION
+  uint64_t CorrelationID = 0;
+  if (xptiTraceEnabled()) {
+    xpti::mem_alloc_data_t MemAlloc{ObjHandle, AllocPtr, 0 /* alloc size */,
+                                    0 /* guard zone */};
+
+    CorrelationID = xptiGetUniqueId();
+    xptiNotifySubscribers(
+        GMemAllocStreamID,
```
Comment (vladimirlaz) [inline sycl/source/detail/memory_manager.cpp:86]: It looks like **GMemAllocStreamID** is not initialized anywhere. is that expected?

**Review by vladimirlaz — APPROVED** [review]

**Comment by alexbatashev** [comment]
Docs build error comes from pulldown

**Pull request merged**